qdev_wrappers/T3/customised_instruments.py

```diff
@@ -243,49 +243,40 @@ def __init__(self, name, address, config, **kwargs):
             delay = float(config_settings[5])
-            self.channels[chan].volt.set_step(step)
-            self.channels[chan].volt.set_delay(delay)
-            
-            
             rangemin = float(config_settings[6])
             rangemax = float(config_settings[7])
-            vldtr = vals.Numbers(rangemin, rangemax)
-            self.channels[chan].volt.set_validator(vldtr)
-            if config_settings[8] == 'fine':
-                if np.mod(chan,4)<3:
-                    self.channels[chan].fine_volt.label = config_settings[1]
-                    self.channels[chan].fine_volt.unit = config_settings[2]
-                    if config_settings[3] not in '1':
-                        exec('self.{} = VoltageDivider(self.channels[chan].fine_volt, {}, label=\'{}\')'.format(config_settings[0],config_settings[3],config_settings[1]))
-                        exec('self.{}.label = \'{}\''.format(config_settings[0],config_settings[1]))
-                        exec('self.{}.unit = \'{}\''.format(config_settings[0],config_settings[2]))
-                        self.channels[chan].fine_volt.division_value = float(config_settings[3])
-                        self.channels[chan].fine_volt._meta_attrs.extend(["division_value"])
-                    else:
-                        exec('self.{} = self.channels[chan].fine_volt'.format(config_settings[0]))
-                        exec('self.{}.label = \'{}\''.format(config_settings[0],config_settings[1]))
-                        exec('self.{}.unit = \'{}\''.format(config_settings[0],config_settings[2]))
-            if config_settings[8] == 'coarse':
-                if np.mod(chan,4)<3:
-                    self.channels[chan].volt.label = config_settings[1]
-                    self.channels[chan].volt.unit = config_settings[2]
-                    if config_settings[3] not in '1':
-                        exec('self.{} = VoltageDivider(self.channels[chan].volt, {}, label=\'{}\')'.format(config_settings[0],config_settings[3],config_settings[1]))
-                        exec('self.{}.label = \'{}\''.format(config_settings[0],config_settings[1]))
-                        exec('self.{}.unit = \'{}\''.format(config_settings[0],config_settings[2]))
-                        self.channels[chan].volt.division_value = float(config_settings[3])
-                        self.channels[chan].volt._meta_attrs.extend(["division_value"])
-                    else:
-                        exec('self.{} = self.channels[chan].volt'.format(config_settings[0]))
-                        exec('self.{}.label = \'{}\''.format(config_settings[0],config_settings[1]))
-                        exec('self.{}.unit = \'{}\''.format(config_settings[0],config_settings[2]))
-
+            fine_mode = config_settings[8]
+
+            if  fine_mode == 'fine':
+                param = channel.fine_volt
+            elif fine_mode == 'coarse':
+                param = channel.fine_volt
```

Comment:
think this should be channel.volt for coarse mode.

Reply:
well spotted.